lib/rubocop/cop/style/multiline_memoization.rb

```diff
@@ -52,13 +52,28 @@ def on_or_asgn(node)
         private
 
         def autocorrect(node)
+          b = node.loc.begin
+          e = node.loc.end
+
           lambda do |corrector|
             if style == :keyword
-              corrector.replace(node.loc.begin, 'begin')
-              corrector.replace(node.loc.end, 'end')
+              replacement = if newline_after?(node, b.end_pos)
+                              'begin'
+                            else
+                              "begin\n" + (' ' * b.begin_pos)
+                            end
+              corrector.replace(b, replacement)
+
+              replacement = if newline_before?(node)
```

Comment:
I don't like reusing variable names - makes it a bit confusing to read the code.